mypyc/irbuild/classdef.py

```diff
@@ -467,12 +467,13 @@ def populate_non_ext_bases(builder: IRBuilder, cdef: ClassDef) -> Value:
     The tuple is passed to the metaclass constructor.
     """
     is_named_tuple = cdef.info.is_named_tuple
+    is_enum = cdef.info.is_enum
     ir = builder.mapper.type_to_ir[cdef.info]
     bases = []
     for cls in cdef.info.mro[1:]:
         if cls.fullname == "builtins.object":
             continue
-        if is_named_tuple and cls.fullname in (
+        if (is_named_tuple or is_enum) and cls.fullname in (
```

Comment:
My thoughts about this change are as the following:

1. `is_extension_class` will return False for `StrEnum` as `metaclass_type` is set to `enum.EnumMeta`  https://github.com/python/mypy/blob/284142d6e29d4588107f5c68f95c43986a2e26e5/mypyc/irbuild/util.py#L130
2. `transform_class_def` mentioned explicitly in comments that classes inheriting from `enum` will have non-extension class object created for them https://github.com/python/mypy/blob/284142d6e29d4588107f5c68f95c43986a2e26e5/mypyc/irbuild/classdef.py#L107
3. For some reason `populate_non_ext_bases` is explicitly checking for named tuple though any class inheriting say `str` or have a `typing.Sequence` in mro it will cause the error currently seen in building code with `StrEnum`  https://github.com/python/mypy/blob/284142d6e29d4588107f5c68f95c43986a2e26e5/mypyc/irbuild/classdef.py#L475

Not really sure if checking against both Enums and Named tuples is generic enough or not.